Public/Src/Cache/ContentStore/Interfaces/Results/ResultsExtensions.cs

```diff
@@ -12,6 +12,12 @@ namespace BuildXL.Cache.ContentStore.Interfaces.Results
     /// </summary>
     public static class ResultsExtensions
     {
+        /// <nodoc />
+        public static string GetDiagnosticsMessageForTracing(this ResultBase result)

```

Comment:
>GetDiagnosticsMessageForTracing [](start = 29, length = 31)

Would it be simpler to add a special getter for `Diagnostics` instead? Then we don't need to remember to use this method in the future. I can guarantee that I won't remember.

Reply:
I'm afraid of a breaking change. Currently `Diagnositcs` is a field and I was not sure if it's safe to make it a property. On the other hand, this extension method is special and gives you an ability to see all the other cases in one place: everyone who calls this method uses `Diagnostics` property to get extra information for successful case (because before this change, `Diagnostics` property was only used for a failure case).

Reply:
I'm afraid of a breaking change. Currently Diagnositcs is a field and I was not sure if it's safe to make it a property. On the other hand, this extension method is special and gives you an ability to see all the other cases in one place: everyone who calls this method uses Diagnostics property to get extra information for successful case (because before this change, Diagnostics property was only used for a failure case).

---
In reply to: [280620200](https://github.com/Microsoft/BuildXL/pull/233#discussion_r280620200) [](ancestors = 280620200)